packages/legend-graph/src/graph-manager/protocol/pure/v1/engine/V1_GraphManagerEngine.ts

```diff
@@ -0,0 +1,419 @@
+import { type V1_RunTestsInput } from './test/V1_RunTestsInput.js';
+import { type V1_RunTestsResult } from './test/V1_RunTestsResult.js';
+import {
+  type V1_MappingModelCoverageAnalysisInput,
+  type V1_MappingModelCoverageAnalysisResult,
+} from './analytics/V1_MappingModelCoverageAnalysis.js';
+import type { ServiceExecutionMode } from '../../../../action/service/ServiceExecutionMode.js';
+import type {
+  V1_CompilationResult,
+  V1_TextCompilationResult,
+} from './compilation/V1_CompilationResult.js';
+import { type V1_GenerateSchemaInput } from './externalFormat/V1_GenerateSchemaInput.js';
+import type { GraphManagerOperationReport } from '../../../../GraphManagerStatistics.js';
+import {
+  type V1_StoreEntitlementAnalysisInput,
+  type V1_DatasetEntitlementReport,
+  type V1_DatasetSpecification,
+  type V1_EntitlementReportAnalyticsInput,
+} from './analytics/V1_StoreEntitlementAnalysis.js';
+import type { V1_SourceInformation } from '../model/V1_SourceInformation.js';
+import type {
+  ClassifierPathMapping,
+  SubtypeInfo,
+} from '../../../../action/protocol/ProtocolInfo.js';
+import { type V1_FunctionActivatorInfo } from './functionActivator/V1_FunctionActivatorInfo.js';
+import { type V1_FunctionActivatorInput } from './functionActivator/V1_FunctionActivatorInput.js';
+import { type V1_RawSQLExecuteInput } from './execution/V1_RawSQLExecuteInput.js';
+import type { V1_ValueSpecification } from '../model/valueSpecification/V1_ValueSpecification.js';
+import {
+  type V1_ArtifactGenerationExtensionOutput,
+  type V1_ArtifactGenerationExtensionInput,
+} from './generation/V1_ArtifactGenerationExtensionApi.js';
+import { type V1_DatabaseToModelGenerationInput } from './relational/V1_DatabaseToModelGenerationInput.js';
+import { type V1_TestDataGenerationInput } from './service/V1_TestDataGenerationInput.js';
+import { type V1_TestDataGenerationResult } from './service/V1_TestDataGenerationResult.js';
+import { type V1_RelationalConnectionBuilder } from './relational/V1_RelationalConnectionBuilder.js';
+import type { PostValidationAssertionResult } from '../../../../../DSL_Service_Exports.js';
+import { type V1_DebugTestsResult } from './test/V1_DebugTestsResult.js';
+import type { TEMPORARY__AbstractEngineConfig } from '../../../../action/TEMPORARY__AbstractEngineConfig.js';
+
+export interface V1_GraphManagerEngine {
+  config: TEMPORARY__AbstractEngineConfig;
+
+  setup: (config: TEMPORARY__EngineSetupConfig) => Promise<void>;
+
+  // ----------------------------------------- Server Client ----------------------------------------
+  /**
+   * NOTE: ideally, we would not want to leak engine server client like this,
+   * since the communication with engine client should only be done in this class
+   * alone. However, we need to expose the client for plugins, tests, and dev tool
+   * configurations.
+   */
+  getCurrentUserId: () => string | undefined;
```

Comment:
I think we should limit these calls, like said, we're abstracting at `engine` level, we could assume the engine does not come with a `server client`. Except for `serverClientGetBaseUrl`, all others are not needed. In fact, they are convenient utilities, we can move those to the server client within each of engine, let's not expose them at interface level like this.

Reply:
I removed these all from the interface, and now wherever they are used I added a `guaranteeType(..., V1_RemoteEngine)` so we can check that only the remote engine is used for things that use the underlying server client implementation.